File: done/1828.py
# 1828 - Bazinga!
rules = [
    ['tesoura','papel'],
    ['papel','tesoura'],
    ['papel','pedra'],
    ['pedra','papel'],
    ['pedra','lagarto'],
    ['lagarto','pedra'],
    ['lagarto','Spock'],
    ['Spock','lagarto'],
    ['Spock','tesoura'],
    ['tesoura','Spock'],
    ['tesoura','lagarto'],
    ['lagarto','tesoura'],
    ['lagarto','papel'],
    ['papel','lagarto'],
    ['papel','Spock'],
    ['Spock','papel'],
    ['Spock','pedra'],
    ['pedra','Spock'],
    ['pedra','tesoura'],
    ['tesoura','pedra']
]

for i in range(1, int(input()) + 1):
    turn = input().split()

    if turn[0] == turn[1]:
        print('Caso #{}: De novo!'.format(i))
    else:
        index = rules.index(turn)
        if index % 2 == 0:
            print('Caso #{}: Bazinga!'.format(i))
        else:
            print('Caso #{}: Raj trapaceou!'.format(i))

File: problems/c/1828.c
#include <stdio.h>
#include <string.h>

int main(void)
{
    int i, t, caso = 1, n;
    scanf("%d", &t);
    char sheldon[8], raj[8];

    while (t--)
    {
        scanf("%s %s", sheldon, raj);

        if (strcmp(sheldon, raj) == 0)
        {
            printf("Caso #%d: De novo!\n", caso);
        }
        else if (strcmp(sheldon, "tesoura") == 0)
        {
            if ((strcmp(raj, "papel") == 0) || (strcmp(raj, "lagarto") == 0))
                printf("Caso #%d: Bazinga!\n", caso);
            else
                printf("Caso #%d: Raj trapaceou!\n", caso);
        }
        else if (strcmp(sheldon, "papel") == 0)
        {
            if ((strcmp(raj, "pedra") == 0) || (strcmp(raj, "Spock") == 0))
                printf("Caso #%d: Bazinga!\n", caso);
            else
                printf("Caso #%d: Raj trapaceou!\n", caso);
        }
        else if (strcmp(sheldon, "pedra") == 0)
        {
            if ((strcmp(raj, "lagarto") == 0) || (strcmp(raj, "tesoura") == 0))
                printf("Caso #%d: Bazinga!\n", caso);
            else
                printf("Caso #%d: Raj trapaceou!\n", caso);
        }
        else if (strcmp(sheldon, "lagarto") == 0)
        {
            if ((strcmp(raj, "Spock") == 0) || (strcmp(raj, "papel") == 0))
                printf("Caso #%d: Bazinga!\n", caso);
            else
                printf("Caso #%d: Raj trapaceou!\n", caso);
        }
        else
        {
            if ((strcmp(raj, "tesoura") == 0) || (strcmp(raj, "pedra") == 0))
                printf("Caso #%d: Bazinga!\n", caso);
            else
                printf("Caso #%d: Raj trapaceou!\n", caso);
        }
        caso++;
    }

    return 0;
}
